### ISSUE TITLE
error when running PhilosopherDbAnnotate

### ISSUE THREAD

**Issue by duselis** [Nesvilab/FragPipe#290]
My search (4 experiments, mostly default parameters for a closed search with Protein and PeptideProphet processing) exits when trying to complete the PhilosopherDbAnnotate process.  Can you give me some direction where I'm going wrong?  Attached are the log file and a truncated fasta (the actual fasta is 138k entries, so not an easy upload.) Thank you!
[trunc_fasta.fas.txt](https://github.com/Nesvilab/FragPipe/files/5796965/trunc_fasta.fas.txt)
[log_2021-01-11_10-42-40.txt](https://github.com/Nesvilab/FragPipe/files/5796966/log_2021-01-11_10-42-40.txt)
[fragpipe_2021-01-11_09-09-44.txt](https://github.com/Nesvilab/FragPipe/files/5796971/fragpipe_2021-01-11_09-09-44.txt)

**Comment by fcyu** [comment]
Hi @prvst, 

Can you take a look when you have time?

Hi @duselis ,

For your search, I suggest you using 1 split database (no split) rather than 99.

Best,

Fengchao

**Comment by prvst** [comment]
Hi @duselis, Philosopher needs to parse the protein FASTA headers to get some information from the entries, things like the gene name, protein existence, and organism. For that to happen, we require that the headers follow certain formats. It looks to me that some of your entries were edited, as they seem to have pieces from different online databases. For the program to work, you will need to change the header to a commonly accepted format, like the ones provided by UniProt, NCBI, or ENSEMBL, or you can adopt a simple custom format (in your case, no spaces and special characters). You can find more information [here](https://github.com/Nesvilab/philosopher/wiki/How-to-Prepare-a-Protein-Database). Please, also make sure that you read the program documentation before using It to avoid any other issues.

**Comment by duselis** [comment]
The original FASTA was downloaded directly from Uniprot.  Decoys and contaminants were added through philosopher v3.3.12, a la [https://github.com/Nesvilab/philosopher/wiki/How-to-Prepare-a-Protein-Database], which produced the above error. The same error occurred when I added decoys and contaminants in the same fashion with philosopher v3.2.3.  

Thank you for pinpointing the error to the FASTA headers though.  I added my own decoys (reversed the sequences and prepended "REV_" to the original headers), which worked fine.  Thanks!
